### ISSUE TITLE
problem with addParameter for pre-R2013b Matlab

### ISSUE THREAD

**Issue by otoolej** [scottclowe/superbar#11]
Hi Scott, a great addition to matlab, and nicely done!

I'm using R2013a, for which `addParameter` is not defined. So to get it working I changed:

-addParameter(parser,  ...
+parser.addParamValue( ...

in `superbar.m`. I have a patch (which just does the above for all instances) but probably would not make sense to apply if matlab are going to phase-out `addParamValue`.

anyhow, thanks for the code.

**Comment by scottclowe** [comment]
Hi @otoolej,

Thanks for the feedback!

I was aware that the code unfortunately does not run on versions R2011a and prior, which is before the introduction of `inputParser`. But was not aware of this problem with R2013a.

I just checked now on my copy of R2016a, and `parser.addParamValue` is valid syntax here too.
And MathWorks is moving MATLAB towards object oriented not away from it, so I don't expect `parser.addParamValue` to be deprecated any time soon. So what we will do is change it to use `parser.addParamValue` everywhere and get broader support.

Would you like to issue a PR for this? I don't mind doing it myself since it is a single find and replace over `superbar.m`, but I thought you might want commit authorship credit for the fix. Let me know either way!

**Comment by otoolej** [comment]
Hi Scott, 
thanks, but much better if you can do it yourself. Just be aware that on the Matlab doc. page for `addParamValue` they issue a NOT RECOMMENDED warning, which could indicate that they may depreciate it at some stage, but who knows with matlab.

again, thanks for the code.

**Comment by scottclowe** [comment]
Okay, thanks for letting me know.

What I'll do then is swap it to `parser.addParamValue` and make a new issue to implement a custom input handling function, which seems like the only way to make something work across a wide range of MATLAB versions.

**Comment by scottclowe** [comment]
Hi @otoolej,
I hadn't realised initially that the problem was the missing `addParameter` method, which has superseded `addParamValue`. (I originally thought the difference was just `obj.method(...)` vs `method(obj ,...)` syntax.)

It seems likely to me then that `addParamValue` will be removed by R2017a, since MathWorks do not seem to keep deprecated functions for long.

What I propose instead is a wrapper which picks whichever of `addParameter` or `addParamValue` is available, like so:

```
parser = inputParser;

if ismethod(parser, 'addParameter');
    safelyAddParameter = @(x) parser.addParameter(x{:});
elseif exist('addParamValue', 'builtin')
    safelyAddParameter = @(x) addParamValue(parser, x{:});
else
    error(...
        ['Could not find a function to add parameters to the input parser.'
        ' Please upgrade your MATLAB installation to R2013a or later.']);
end

safelyAddParameter({'Orientation', 'v', @ischar});
```

Could you check that code block works in R2013a for me?

**Comment by otoolej** [comment]
Hi Scott, 

I think you meant the `parser.addParameter` / `addParamValue(parser` the other way around:

```
parser = inputParser;

if exist( 'addParameter','builtin')
    safelyAddParameter = @(x) addParameter(parser,x{:});
elseif ismethod(parser,'addParamValue')
    safelyAddParameter = @(x) parser.addParamValue(x{:});
else
    error(...
        ['Could not find a function to add parameters to the input parser.' ...
        ' Please upgrade your MATLAB installation to R2013a or later.']);
end
```

... definitely confusing but the above is working with my r2013a matlab -- but maybe not for you?

**Comment by scottclowe** [comment]
Issue was resolved in commit e6d8239f8.
